### PR TITLE
Jbaker/UI updates

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #141, #144 

## Summary:

- Fixes inconsistent doc generation
- Align the day numbers when viewed on mobile iOS
- Fixes up :hover styles so they don't apply on mobile devices
- Fixes the date calculation of test so it works correctly on the last day of the month

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/style-auro-calendar-cell.scss [hunk 1]
```diff
@@ -54,6 +60,7 @@
   &:hover {
     background-color: #f0f0f0; // var(--auro-color-ui-bkg-hover-on-light); - this token was part of the design but caused a problem with the `inRange` date styles so we had to switch to a solid color for hover.
     color: var(--auro-color-text-primary-on-light);
+
```
Comment (jordanjones243) [inline src/style-auro-calendar-cell.scss:68]: I am not seeing a fix for the calendar gradient listed in ticket #141

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
There are two commits with the same message, they need to be squashed

File: src/style-auro-calendar-cell.scss (hunk 1 above)
Comment (jason-capsule42) [reply src/style-auro-calendar-cell.scss:68]: That part is not getting handled in this PR. There is a full review of the mobile that is going to happen.

**Review by jordanjones243 — DISMISSED** [review]

File: src/style-auro-calendar-cell.scss (hunk 1 above)
Comment (blackfalcon) [reply src/style-auro-calendar-cell.scss:64]: Right, work moved to this issue,  see https://github.com/AlaskaAirlines/auro-datepicker/issues/141#issuecomment-1702114839

- #140

File: src/style-auro-calendar.scss [hunk 2]
```diff
@@ -6,38 +6,39 @@
...
   background-color: transparent;
   color: var(--auro-color-border-active-on-light);
-  cursor: pointer;
 
-  &:hover {
-    border-color: var(--auro-color-border-active-on-light);
-  }
+  cursor: pointer;
 
   auro-icon {
```
Comment (blackfalcon) [inline src/style-auro-calendar.scss:39]: There are still a lot of references to `auro-icon` that are not null that we are not using that element. 

```
style-auro-calendar.scss
39:  auro-icon {

auro-calendar.js
339:            <auro-icon category="interface" name="chevron-left" customColor></auro-icon>

style-auro-calendar-month.scss
75:  auro-icon {
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**